django/core/files/storage.py

```diff
@@ -425,3 +433,56 @@ def _setup(self):
 
 
 default_storage = DefaultStorage()
+
+
+class InvalidStorageError(ImproperlyConfigured):
+    pass
+
+
+class StorageHandler:
+    def __init__(self, settings=None):
+        """
+        settings is an optional dict of storage backend definitions
+        (structured like settings.STORAGES).
+        """
+        self._settings = settings
+        self._storages = {}
+
+    @cached_property
+    def settings(self):
+        if self._settings is None:
+            self._settings = django_settings.STORAGES
+
+        self._settings.setdefault(
+            DEFAULT_STORAGE_ALIAS, {"BACKEND": django_settings.DEFAULT_FILE_STORAGE}
+        )
+        self._settings.setdefault(
+            STATICFILES_STORAGE_ALIAS, {"BACKEND": django_settings.STATICFILES_STORAGE}
+        )
```

Comment:
IMO default values should be set in `LazySettings`/`Settings` (`django/conf/__init__.py`) and we should immediately deprecate `DEFAULT_FILE_STORAGE` and `STATICFILES_STORAGE` settings (see e.g. 226ebb17290b604ef29e82fb5c1fbac3594ac163).

Also, the existing code outside of `django/conf/__init__.py` cannot use deprecated settings.

Reply:
Moved STORAGES setting defaults to `django/conf/__init__.py` and `django/conf/global_settings.py`. I'm still working on removing references to deprecated settings outside of `django/conf/__init__.py`.